pkg/controllers/cassandra/nodepool/resource.go

```diff
@@ -35,6 +35,17 @@ func StatefulSetForCluster(
 	statefulSetName := util.NodePoolResourceName(cluster, np)
 	seedProviderServiceName := util.SeedProviderServiceName(cluster)
 	nodePoolLabels := util.NodePoolLabels(cluster, np.Name)
+
+	datacenter := "navigator-default-datacenter"
+	if np.Datacenter != "" {
+		datacenter = np.Datacenter
+	}
+
+	rack := "navigator-default-rack"
+	if np.Rack != "" {
+		rack = np.Rack
+	}
```

Comment:
Would it make sense to default 'rack' to the node pool name? The only problem with that is it requires explicit action from the user to *disable* the rack awareness (they'd need to explicitly set 'rack' to some static value on all node pools)

Reply:
This sounds reasonable. I'd imagine if a user didn't care about rack awareness they would only specify a single nodepool resource.

Reply:
I'm not 100% sold though, as it may be surprising to a user that has never specified a `rack`, `datacenter` or `nodeSelector` including a zone attribute to find that their data is being spread in a rack-aware fashion.

Spreading across node pools by default isn't really that desirable if the properties of that node pool are relatively similar (e.g. spread across all zones in the cluster due to lack of nodeSelector).

cc @wallrj

Reply:
I'm not really sure.
 * I think using nodepool name as rack name is reasonable until we understand it better.
 * If you don't specify a `nodeSelector` the nodepool pods and their PVs could end up in any zone. And once those PVs are created it probably means that the pod is stuck in that zone (on AWS and GCP at least). 
So the nodeselector should probably be immutable too.

Reply:
In fact my preference would be to remove the defaults and make these required fields....but I understand if you decide to go with the defaults for now.